### PR TITLE
Implement SDL-0223 Add Currently Playing Media Image to MediaServiceData

### LINKED ISSUE #1247: [SDL 0223] Add Currently Playing Media Image to MediaServiceData
Proposal: [Add Currently Playing Media Image to MediaServiceData](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0223-media-service-image.md)

> This proposal adds an image for the currently playing media item to the media app service.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/706

### PR DESCRIPTION
**Pull request by justingluck93**
Fixes #1247

This PR is **[ready]** for review.

### Risk
This PR makes **[minor]** API changes.

### Summary
Added an image parameter to the data 

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/SDLMediaServiceData.h [hunk 1]
```diff
@@ -35,7 +36,17 @@ NS_ASSUME_NONNULL_BEGIN
...
+- (instancetype)initWithMediaType:(nullable SDLMediaType)mediaType mediaImage:(nullable SDLImage *)mediaImage mediaTitle:(nullable NSString *)mediaTitle mediaArtist:(nullable NSString *)mediaArtist mediaAlbum:(nullable NSString *)mediaAlbum playlistName:(nullable NSString *)playlistName isExplicit:(BOOL)isExplicit trackPlaybackProgress:(UInt32)trackPlaybackProgress trackPlaybackDuration:(UInt32)trackPlaybackDuration queuePlaybackProgress:(UInt32)queuePlaybackProgress queuePlaybackDuration:(UInt32)queuePlaybackDuration queueCurrentTrackNumber:(UInt32)queueCurrentTrackNumber queueTotalTrackCount:(UInt32)queueTotalTrackCount;
```
Comment (NicoleYarroch) [inline SmartDeviceLink/SDLMediaServiceData.h]: Modifying the convenience init like this is a breaking change. Instead you should deprecate the current init and create another init with the new parameter. Check out `SDLGetVehicleData.h` to see an example of deprecating a method. Also, don't forget to add documentation for the new parameter to the method documentation.

Comment (justingluck93) [reply SmartDeviceLink/SDLMediaServiceData.h]: fixed

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLMediaServiceDataSpec.m [hunk 2]
```diff
@@ -101,8 +108,9 @@
...
-        SDLMediaServiceData *testStruct = [[SDLMediaServiceData alloc] initWithMediaType:testMediaType mediaTitle:testMediaTitle mediaArtist:testMediaArtist mediaAlbum:testMediaAlbum playlistName:testPlaylistName isExplicit:testIsExplicit trackPlaybackProgress:testTrackPlaybackProgress trackPlaybackDuration:testTrackPlaybackDuration queuePlaybackProgress:testQueuePlaybackProgress queuePlaybackDuration:testQueuePlaybackDuration queueCurrentTrackNumber:testQueueCurrentTrackNumber queueTotalTrackCount:testQueueTotalTrackCount];
+        SDLMediaServiceData *testStruct = [[SDLMediaServiceData alloc] initWithMediaType:testMediaType mediaImage:testMediaImage mediaTitle:testMediaTitle mediaArtist:testMediaArtist mediaAlbum:testMediaAlbum playlistName:testPlaylistName isExplicit:testIsExplicit trackPlaybackProgress:testTrackPlaybackProgress trackPlaybackDuration:testTrackPlaybackDuration queuePlaybackProgress:testQueuePlaybackProgress queuePlaybackDuration:testQueuePlaybackDuration queueCurrentTrackNumber:testQueueCurrentTrackNumber queueTotalTrackCount:testQueueTotalTrackCount];
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLMediaServiceDataSpec.m]: added new test

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLMediaServiceDataSpec.m [hunk 3]
```diff
@@ -100,10 +107,32 @@
         expect(testStruct.queueTotalTrackCount).to(equal(testQueueTotalTrackCount));
     });
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
     it(@"Should get correctly when initialized with initWithMediaType:mediaTitle:mediaArtist:mediaAlbum:playlistName:isExplicit:trackPlaybackProgress:trackPlaybackDuration:queuePlaybackProgress:queuePlaybackDuration:queueCurrentTrackNumber:queueTotalTrackCount:", ^{
         SDLMediaServiceData *testStruct = [[SDLMediaServiceData alloc] initWithMediaType:testMediaType mediaTitle:testMediaTitle mediaArtist:testMediaArtist mediaAlbum:testMediaAlbum playlistName:testPlaylistName isExplicit:testIsExplicit trackPlaybackProgress:testTrackPlaybackProgress trackPlaybackDuration:testTrackPlaybackDuration queuePlaybackProgress:testQueuePlaybackProgress queuePlaybackDuration:testQueuePlaybackDuration queueCurrentTrackNumber:testQueueCurrentTrackNumber queueTotalTrackCount:testQueueTotalTrackCount];
+#pragma clang diagnostic pop
+
+        expect(testStruct.mediaType).to(equal(testMediaType));
```
Comment (NicoleYarroch) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLMediaServiceDataSpec.m]:
```suggestion
        expect(testStruct.mediaImage).to(beNil());
        expect(testStruct.mediaType).to(equal(testMediaType));
```

Comment (justingluck93) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLMediaServiceDataSpec.m]: is line 161 not enough?

Comment (justingluck93) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLMediaServiceDataSpec.m]: I do this on line 128

**Pull request merged**